src/sklearndf/wrapper/_wrapper.py

```diff
@@ -448,14 +449,28 @@ def _post_fit(
             self._outputs = y.columns.tolist()
 
     # noinspection PyPep8Naming
-    def _check_parameter_types(
+    def _validate_parameter_types(
         self,
-        X: pd.DataFrame,
+        X: Union[pd.Series, pd.DataFrame],
         y: Optional[Union[pd.Series, pd.DataFrame]],
         *,
         expected_columns: pd.Index = None,
-    ) -> None:
-        if not isinstance(X, pd.DataFrame):
+    ) -> Tuple[pd.DataFrame, Union[pd.Series, pd.DataFrame]]:
+        # Check that the X and y parameters are valid data frames and series,
+        # and return X as a data frame and y as a series or data frame.
+        #
+        # If X is a series, convert it to a data frame with a single column.
+        #
+        # If expected_columns is not None, check that the columns of X match
+        # the expected columns.
+
+        if isinstance(X, pd.Series):
+            if X.name is None:
+                raise ValueError(
+                    "the name of the series passed as arg X must not be None"
+                )
+            X = X.to_frame()
+        elif not isinstance(X, pd.DataFrame):
             raise TypeError("arg X must be a DataFrame")
```

Comment:
Just a nitpick: `"arg X must be a DataFrame or a Series"`?

Reply:
Indeed - it should be!